File: vipimo/heshi/bench.go
package main

import "fmt"

const idadi = 2000000

func heshiFnv(s []byte, urefu int32) int32 {
	h := int32(-2128831035)
	for i := int32(0); i < urefu; i++ {
		h ^= int32(s[i])
		h *= 16777619
	}
	return h
}

func main() {
	neno := []byte("kompyuta_ya_mfano_kwa_uzito_wa_kati_12345")
	urefu := int32(len(neno))
	var jumla int64 = 0
	for i := int64(0); i < idadi; i++ {
		h := heshiFnv(neno, urefu)
		jumla += int64(h) + i
	}
	fmt.Printf("jumla=%d urefu=%d\n", jumla, urefu)
}

File: vipimo/heshi/bench.c
#include <stdio.h>
#include <string.h>

#define IDADI 2000000

static int heshi_fnv(const char* s, int urefu) {
    int h = -2128831035;
    for (int i = 0; i < urefu; i++) {
        h = h ^ (unsigned char)s[i];
        h = h * 16777619;
    }
    return h;
}

int main() {
    const char* neno = "kompyuta_ya_mfano_kwa_uzito_wa_kati_12345";
    int urefu = (int)strlen(neno);
    long long jumla = 0;
    for (int i = 0; i < IDADI; i++) {
        int h = heshi_fnv(neno, urefu);
        jumla += h + i;
    }
    printf("jumla=%lld urefu=%d\n", jumla, urefu);
    return 0;
}
